dags/map_reproducibility/benchmarkdb_utils.py

```diff
@@ -0,0 +1,263 @@
+    xla_flags: str = "",
+    topology: str = "",
+    dataset: str = "",
+    num_of_superblock: int = None,
+    update_person_ldap: str = getpass.getuser(),
+    comment: str = "",
+    is_test: bool = False,
+):
+  """Writes a workload benchmark run manually to the database.
+
+  This function validates the provided IDs and, if valid, constructs a
+  WorkloadBenchmarkV2Schema object with the given data and writes it to the
+  "run_summary" table in BigQuery.
+
+  Args:
+    model_id: The ID of the model used in the run.
+    hardware_id: The ID of the hardware used in the run.
+    software_id: The ID of the software stack used in the run.
+    number_of_nodes: The number of nodes used in the run.
+    number_of_chips: The number of chips used in the run.
+    container_image_name: The name of the container image used in the run.
+    global_batch_size: The global batch size used in the run.
+    precision: The precision used in the run (e.g., fp32, bf16).
+    optimizer: The optimizer used in the run (e.g., adam, sgd).
+    seq_length: The sequence length used in the run.
+    median_step_time: The median step time of the run.
+    e2e_time: The end-to-end time of the run.
+    number_of_steps: The number of steps taken in the run.
+    mfu: The MFU (model flops utilization) achieved in the run.
+    tokens_per_second: The tokens per second achieved in the run.
+    run_type: The type of run (default: "perf_optimization").
+    run_release_status: possible values "local" ( code changes are done locally), "prep_release" ( all code code changes are present in the image)
+    other_metrics_in_json: A JSON string containing other metrics.
+    nccl_driver_nickname: The nickname of the NCCL driver used.
+    env_variables: A string containing environment variables.
+    framework_config_in_json: A JSON string containing framework configurations.
+    xla_flags: A json string containing all the XLA flags.
+    topology: The topology of the hardware used in the run. ( valid for TPUs)
+    dataset: The dataset used in the run.
+    num_of_superblock: The number of superblocks in the hardware. ( valid for GPUs)
+    update_person_ldap: The LDAP ID of the person updating the record (default: current user).
+    comment: A comment about the run.
+    is_test: Whether to use the testing project or the production project.
+
+  Raises:
+    ValueError: If any of the IDs are invalid.
+  """
+
+  sys.path.append(writer_path)
+
+  # pylint: disable=import-outside-toplevel
+  import logging
+  import uuid
+  from typing import Type
+
+  from aotc.benchmark_db_writer import bq_writer_utils
+  from aotc.benchmark_db_writer.schema.workload_benchmark_v2 import workload_benchmark_v2_schema
+  from aotc.benchmark_db_writer.schema.workload_benchmark_v2 import model_info_schema
+  from aotc.benchmark_db_writer.schema.workload_benchmark_v2 import software_info_schema
+  from aotc.benchmark_db_writer.schema.workload_benchmark_v2 import hardware_info_schema
```

Comment:
qq: can we move those import to the top level, any specific reason blocking us so we have to keep inside the functions?